src/kde/quick_index.hh

```diff
@@ -0,0 +1,145 @@
+//--------------------------------------------*-C++-*---------------------------------------------//
+/*!
+ * \file   kde/quick_index.hh
+ * \author Mathew Cleveland
+ * \brief  This class generates coarse spatial indexing to quickly access near-neighbor data. This
+ *         additionally provides simple interpolation schemes to map data to simple structured
+ *         meshes. 
+ * \note   Copyright (C) 2018-2021 Triad National Security, LLC.
+ *         All rights reserved. */
+//------------------------------------------------------------------------------------------------//
+
+#ifndef rtt_kde_quick_index_hh
+#define rtt_kde_quick_index_hh
+
+#include "c4/global.hh"
+#include "units/MathConstants.hh"
+#include <array>
+#include <map>
+#include <vector>
+
+namespace rtt_kde {
+
+//================================================================================================//
+/*!
+ * \brief quick_index
+ *
+ * Provide a hash like index of spatial distributed data along with simple mapping functions.
+ * 
```

Comment:
You say `hash-like`, can you actually use `std::hash` (https://en.cppreference.com/w/cpp/utility/hash)?